File: leetcode/go/2595_number_of_even_and_odd_bits.go
package main

func evenOddBit(n int) []int {
	i, even, odd := 0, 0, 0
	for n != 0 {
		if n%2 == 1 {
			if i%2 == 1 {
				odd++
			} else {
				even++
			}
		}
		i++
		n /= 2
	}
	return []int{even, odd}
}

// github.com/p-nerd (Shihab Mahamud)
// Wednesday, April 17, 2024 | 11:39:21 PM (+6)

File: leetcode/cpp/2595_number_of_even_and_odd_bits.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
  public:
    vector<int> evenOddBit(int n)
    {
        int i = 0, even = 0, odd = 0;
        while (n)
        {
            int bit = n % 2;
            if (bit == 1)
            {
                if (i % 2 == 0)
                {
                    even++;
                }
                else
                {
                    odd++;
                }
            }
            i++;
            n /= 2;
        }
        return {even, odd};
    }
};

int main()
{

    return 0;
};
// github.com/p-nerd (Shihab Mahamud)
// Wednesday, April 17, 2024 | 11:22:57 PM (+6)
